### PR TITLE
Fixing Migration crashes on MySQL/MariaDB & PostgreSQL

### PR DESCRIPTION
**Pull request by josegomezr**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [ ] I have added necessary documentation (if appropriate).

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

- Fixes Issue #2422 regarding migration crashes on Postgresql because of incompatible type changes, and mysql because of uuid fields.
- Maybe not so rails-lish? But it works

**Changes proposed in this pull request:**

- Minor changes on migration chain, I've made tests and in my computer works <sup>tm</sup>. I believe these changes are not so drastic. 
- Basically it optionally provides a `using:` parameter for postgresql but not the others.
- Same logic with MySQL uuid fields.

### CODE REVIEW

**Comment by hennevogel** [comment]
See #2422

**Comment by hennevogel** [comment]
@josegomezr sorry that I had to close this. next time, if you're unsure if we would consider something a bug, you should wait for feedback on the issue before sending a PR. We have clarified this now in our contribution guidelines.

**Comment by josegomezr** [comment]
Noted. No problem.
